panels/models/attraction.py

```diff
@@ -444,6 +562,8 @@ def available_events_by_day(self):
 # =====================================================================
 class AttractionEvent(MagModel):
     attraction_feature_id = Column(UUID, ForeignKey('attraction_feature.id'))
+    attraction_id = Column(UUID, ForeignKey('attraction.id'), index=True)
```

Comment:
This is a design pattern I am testing out. The `Attractions` hierarchy looks like this:
```
Attraction
  + AttractionFeature
      + AttractionEvent
          + AttractionSignup
          + AttractionNotification
          + AttractionNotificationReply
```

I found I was doing a lot of expensive joins and traversals to get from a child object back to the `Attraction` object. I denormalized the `attraction_id`, so I could reach/join from any of the child objects directly to the parent `Attraction`.

This does have performance benefits, but it also requires extra attention to make sure the `attraction_id` is set correctly. This is mostly taken care of by using `AssociationProxys` and `@presave_adjustments`. But it's something to be aware of.